lib/util.ml

```diff
@@ -28,32 +29,94 @@ let sign_string_sha256 ~key ~basestring =
   Cstruct.of_string basestring |> Nocrypto.Hash.SHA256.hmac ~key:(Cstruct.of_string key) |> Hex.of_cstruct |> Hex.show
 
 module Build = struct
+  type pipeline_check = {
+    is_pipeline_step : bool;
+    pipeline_name : string;
+  }
+
   let buildkite_is_failed_re = Re2.create_exn {|^Build #\d+ failed|}
 
+  let parse_context ~context ~build_url =
```

Comment:
This and `get_build_number` below do a lot of string manipulation just to obtain the pipeline name. Can it be simpler? It reads like the function wants to target Buildkite mainly but also somewhat support status contexts in general.
Instead if there is an interface to extract the necessary build info, and we implement a `Buildkite` module that programs to that interface, then we expect a predictable structure for both build context and url, and can extract the info in 3 lines instead of 30